corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py

```diff
@@ -364,19 +365,23 @@ def update_log(action, doc_ids):
                     obj.save()
         update_log("Created", [obj._migration_couch_id for obj in creates])
         update_log("Updated", [obj._migration_couch_id for obj in updates])
-        update_log("Ignored", [doc["_id"] for doc in ignored])
-        self.ignored_count += len(ignored)
+        update_log("Ignored", ignored_ids)
+        self.ignored_count += len(ignored_ids)
+        return set(ignored_ids)
```

Comment:
I think it'd be helpful to have some kind of comment talking about ignored documents to orient developers who are subclassing this comment. Maybe function-level comments on `_migrate_docs` and `_migrate_doc`, or maybe just a bit more context on the comment for `should_ignore`. I think they look more prominent in the code (because they're used as a return value here) than they really are.

To check my understanding, ignored docs are only for situations where couch created multiple docs but only one of them is ever accessed, right?

Reply:
> ignored docs are only for situations where couch created multiple docs but only one of them is ever accessed, right?

Not quite. The `should_ignore` hook was required for `LookupTableRow` (`FixtureDataItem`) and `LookupTableRowOwner` (`FixtureOwnership`) because there are documents in Couch that have been orphaned from their parent (the parent doc has been deleted), and they cannot be inserted into SQL because a foreign key constraint would fail.

By default `should_ignore` simply returns `False`. It can be overridden if needed, but it's not required.

I'll see if I can think of a way to improve the documentation around this. I realize that I have overridden `_migrate_docs` in one place, but I do not expect that to be done often.